### ISSUE TITLE
Datacenter diversity

### ISSUE THREAD

**Issue by marcovelon** [simplex-chat/simplex-chat#4762]
I've looked up SimpleX Chat server locations from https://status.simplex.chat and have a few suggestions.

**SimpleX Chat currently uses one single hosting provider for all its all servers** which is Linode (belongs to Akamai) in its three EMEA datacenter locations: London (15+ virtual servers), Frankfurt (5 virtual servers in the same datacenter) and Stockholm (5 virtual servers in the same datacenter):

```
smp1.simplex.im 	88.80.191.111	(Linode, London)
smp4.simplex.im 	85.159.210.201	(Linode, London)
smp5.simplex.im 	194.195.243.62	(Linode, Frankfurt)
smp6.simplex.im 	172.232.152.8	(Linode, Stockholm)
smp8.simplex.im 	194.195.243.87	(Linode, Frankfurt)
smp9.simplex.im 	172.232.152.16	(Linode, Stockholm)
smp10.simplex.im 	172.232.147.214	(Linode, Stockholm)
smp11.simplex.im 	139.177.177.148	(Linode, Frankfurt)
smp12.simplex.im 	139.162.197.202	(Linode, London)
smp14.simplex.im 	213.219.37.83	(Linode, London)
smp15.simplex.im 	172.104.138.143	(Linode, Frankfurt)
smp16.simplex.im 	172.104.138.186	(Linode, Frankfurt)
smp17.simplex.im 	172.232.146.28	(Linode, Stockholm)
smp18.simplex.im 	172.232.146.243	(Linode, Stockholm)
smp19.simplex.im 	139.162.221.19	(Linode, London)

xftp1.simplex.im 	143.42.99.82	(Linode, London)
xftp2.simplex.im 	143.42.99.154	(Linode, London)
xftp3.simplex.im 	143.42.99.210	(Linode, London)
xftp4.simplex.im 	213.168.251.135 (Linode, London)
xftp5.simplex.im 	213.168.251.142 (Linode, London)
xftp6.simplex.im 	178.79.132.8	(Linode, London)

ntf2.simplex.im 	139.162.221.251	(Linode, London)
ntf3.simplex.im 	139.162.220.192	(Linode, London)
ntf4.simplex.im 	139.162.249.101	(Linode, London)

turn.simplex.im		139.162.253.141	(Linode, London)

stun.simplex.im		88.80.185.137	(Linode, London)
```

While I understand that most of these are just servers like SMP proxy that transmit encrypted messages and it's unlikely they could produce any useful information during a possible law enforcement intervention, there are still other reasons on why it's better to diversify ISPs and datacenters:

Connectivity: while I admit that Akamai has a global interconnected network with great reliability, there are still minimal risks associated with possible incidents in any of three locations related to (a) data center buildings they are in, (b) Akamai's or their upstream core routers, (c) ISP-wide blacklisting possibly addressed to both directions - from Akamai to SimpleX users (that don't use Tor) ISP IP ranges and vice versa, and (d) possible ISP/project-targeted DDoS attacks.

Easy shutdown: having all servers located represents some certain paradise for LE agencies when they want to shutdown a whole operation of some project. In case it's an issue, all the LE rep needs to do is to send a single email to Akamai's legal compliance department with a takedown request, which would lead to shutdown of all servers immediately with all SimpleX users (that don't use Tor) losing access to the SMP proxy network.

There is yet another less relevant but still important remark I would like to add, given that all Akamai servers are fully virtualized containers (not dedicated hardware) and just in case you don't know, getting a full operational copy of RAM and disk without shutting down or rebooting a virtual server (any virtualization standard) is a matter of 5 minutes without a customer being even able to notice. Most hypervisors allow RAM and disk dumps of its running virtual servers with just a few clicks (see Proxmox GUI for example). This is why bare metal infrastructure is strongly advised for any kind of privacy-oriented projects and given your good funding lately I think it's possible to afford these, given that you can get cheap dedicated servers nowadays from providers like OVH and Hetzner.

I would recommend switching hosting providers for at least 2/3 of the current SMP servers in spite of network and datacenter diversity. Good options are AWS, Hetzner, DataPacket, Zare or colocation of your own hardware across popular datacenters around EMEA. I would also recommend opting by reducing a number of servers per ISP/location, from 5+ to 1-2 with servers paid annually or biannually in all locations for reduced exposure to procedures related to payments. 

Of course the current infrastructure you have right now is totally fine and this doesn't seem like an urgent matter at all given the current project's popularity, but rather a subject for considetation for future growth of SimpleX Chat popularity and userbase.

**Comment by jermanuts** [comment]
I agree, it is a bit better to use different VPSs as they are forced to do "lawful interception" like an actual case where it was done by  Hetzner and Linode targeting the largest Russian XMPP server https://notes.valdikss.org.ru/jabber.ru-mitm/.

It would be much better if these servers where actually hosted by SimpleX Chat team local machines which they physically control.

**Comment by DarkmatterUAE** [comment]
Also there's the risk of BGP hijacking, in which case a BGP autonomous system (AS) wrongly announced they own an IP range it doesn't actually own (whether incidental or deliberate) and corrupted the adjacent ASes' BGP routing table, causing services in that IP range to be unreachable by end users. There has been some high profile incident (or not?) in recent years:
https://en.wikipedia.org/wiki/BGP_hijacking
